### PR TITLE
[extended-monitoring] Restart if metrics were last collected more than 15 minutes

### PR DESCRIPTION
**Pull request by nabokihms**
Signed-off-by: m.nabokikh <maksim.nabokikh@flant.com>

## Why do we need it, and what problem does it solve?
Sometimes extended monitoring exporter can stop sending metrics because of Kubernetes API server connection loss of something.  This PR adds the check that new metrics are collected. Otherwise, kubelet will restart the pod because of the failed liveness probe.

## Checklist
- [ ] The code is covered by unit tests.
- [x] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: extended-monitroing
type: fix
summary: Restart if metrics were last collected more than 15 minutes
impact_level: default
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @nabokihms. I've started the workflow for label 'e2e/run/gcp'...

File: modules/340-extended-monitoring/images/extended-monitoring-exporter/src/extended-monitoring.py [hunk 1]
```diff
@@ -301,8 +304,10 @@ def do_GET(self):
             return
 
         if self.path == "/healthz":
-            self.send_response(200)
+            # Fail if metrics were last collected more than 15 minutes ago
+            self.send_response(200 if self.__class__._last_observe > (datetime.now() - timedelta(minutes=15)) else 400)
```
Comment (zuzzas) [inline modules/340-extended-monitoring/images/extended-monitoring-exporter/src/extended-monitoring.py:308]:
```suggestion
            self.send_response(200 if self.__class__._last_observe > (datetime.now() - timedelta(minutes=15)) else 500)
```

That is not a Bad Request, there is something wrong with the server.

https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#client_error_responses

**Review by zuzzas — CHANGES_REQUESTED** [review]

File: modules/340-extended-monitoring/images/extended-monitoring-exporter/src/extended-monitoring.py (hunk 1 above)
Comment (nabokihms) [reply modules/340-extended-monitoring/images/extended-monitoring-exporter/src/extended-monitoring.py:308]: Ok, it makes sense.

File: modules/340-extended-monitoring/images/extended-monitoring-exporter/src/extended-monitoring.py [hunk 2]
```diff
@@ -301,8 +304,10 @@ def do_GET(self):
             return
 
         if self.path == "/healthz":
-            self.send_response(200)
+            # Fail if metrics were last collected more than 15 minutes ago
+            self.send_response(200 if self.__class__._last_observe > (datetime.now() - timedelta(minutes=15)) else 500)
```
Comment (shvgn) [inline modules/340-extended-monitoring/images/extended-monitoring-exporter/src/extended-monitoring.py:308]: For the sake of readability of the decision making

```suggestion
            fresh_enough = self.__class__._last_observe > (datetime.now() - timedelta(minutes=15))
            self.send_response(200 if fresh_enough else 500)
```

**Review by shvgn — DISMISSED** [review]
Despite a small nitpick, LGTM

**Pull request merged**